### ISSUE TITLE
Add possibility for specifying the distribution type

### ISSUE THREAD

**Issue by vnastasi** [gradle/wrapper-upgrade-gradle-plugin#278]
Hello

I noticed that there is no option in the plugin configurations to specify the distribution type for the wrapper (`all` or `bin`), and the `wrapper` task invoked always defaults to `bin` distribution type.

While this probably is OK for the vast majority of users, for those of us using `all` it's a bit inconvenient. I personally like to dig deeper in Gradle sources in my projects when I make custom tasks and plugins, so having the full distribution is a must. I can always go and amend the resulting PR, of course, but still. A property in the plugin extension would be a lot nicer.

Cheers!

**Comment by StefMa** [comment]
I didn't find this request too bad and think about adding this feature. However, right now the extension or the specific domain object is used for Gradle and Maven. Since this is a gradle specific request we may wanna talk about the api here.
One approach would be to extract the current domain object to an interface and inherit this by specific GradleDomainObject and a MavenGradleObject.

What do you think about this @alextu ?
Not only about the request itself but also about my api approach?

**Comment by alextu** [comment]
Yes, I think that's a reasonable feature. Since the "distribution type" [exists as well for Maven](https://maven.apache.org/wrapper/maven-wrapper-plugin/wrapper-mojo.html#distributionType), we can implement this attribute for both Gradle and Maven in the same domain object.
We won't have time to work on this feature in the near future though, but feel free to submit a PR.

**Comment by StefMa** [comment]
But the type is 100% (except of `bin`) different. Doesn't it make sense in terms of runtime safety to introduce hardcoded enums? So that for Gradle only all and bin is possible while for maven bin source.. etc?

**Comment by StefMa** [comment]
I just checked the code and found out that for Gradle, it fetches https://services.gradle.org/versions/current to find out the latest version and the checksum.
It seems that there is only a checksum for the `bin` vesion, but no checksum of the `all` version.
I get of course see the checksum here https://gradle.org/release-checksums/, but seems I'm unable to get them via json? 🤔 

This would of course a big blocker for this feature...

**Comment by StefMa** [comment]
All the cheksums can also be found here:
https://services.gradle.org/distributions/

**Comment by joe10832** [comment]
https://github.com/gradle/wrapper-upgrade-gradle-plugin/issues/278

**Comment by clayburn** [comment]
This project is no longer maintained and the repository is now archived.

Wrapper upgrades are now available from the two mainstream dependency bots, so the plugin no longer adds value:

- **[Renovate](https://docs.renovatebot.com/)** upgrades both wrappers. Enable the [`gradle-wrapper`](https://docs.renovatebot.com/modules/manager/gradle-wrapper/) and [`maven-wrapper`](https://docs.renovatebot.com/modules/manager/maven-wrapper/) managers.
- **[Dependabot](https://docs.github.com/en/code-security/dependabot)** upgrades the Gradle wrapper through the `gradle` ecosystem. It does not yet upgrade the Maven wrapper, which [dependabot-core#13183](https://github.com/dependabot/dependabot-core/issues/13183) tracks.

Thank you for the report. This issue is closed as not planned.
